tsl/src/bgw_policy/job_api.c

```diff
@@ -161,7 +163,19 @@ job_add(PG_FUNCTION_ARGS)
 	ts_bgw_job_validate_job_owner(owner);
 
 	/* Next, insert a new job into jobs table */
-	namestrcpy(&application_name, "User-Defined Action");
+	if (job_name_str)
+	{
+		/* 7 bytes are reserved for job id which will be appended later. */
+		if (strlen(job_name_str) >= NAMEDATALEN - 7)
+		{
+			ereport(ERROR,
+					(errcode(ERRCODE_NAME_TOO_LONG),
+					 errmsg("application name \"%s\": too long.", job_name_str)));
+		}
+		namestrcpy(&application_name, job_name_str);
```

Comment:
What happens if the job_id is greater than 9999?

Reply:
Moved this check into `ts_bgw_job_insert_relation()` where we know the exact job id.